src/mlx/warnings/regex_checker.py

```diff
@@ -107,12 +161,156 @@ def check(self, content):
         '''
         matches = re.finditer(self.pattern, content)
         for match in matches:
-            if (match.group('curr') == match.group('max')) and \
-                    (match.group('classification') in self.CLASSIFICATION):
-                self.count += 1
-                match_string = match.group(0).strip()
-                self.counted_warnings.append(match_string)
-                self.print_when_verbose(match_string)
+            if (classification := match.group("classification").lower()) in self.checkers:
+                self.checkers[classification].check(match)
+            else:
+                checker = CoverityClassificationChecker(classification=classification, verbose=self.verbose)
+                self.checkers[classification] = checker
+                checker.cq_enabled = self.cq_enabled
+                checker.exclude_patterns = self.exclude_patterns
+                checker.cq_description_template = self.cq_description_template
+                checker.cq_default_path = self.cq_default_path
+                checker.check(match)
+
+    def parse_config(self, config):
+        """Process configuration
+
+        Args:
+            config (dict): Content of configuration file
+        """
+        for key in config:
+            if key == "enabled":
+                continue
+            if key == "cq_description_template":
+                self.cq_description_template = Template(config['cq_description_template'])
+                continue
+            if key == "cq_default_path":
+                self.cq_default_path = config['cq_default_path']
+                continue
+            if key == "exclude":
+                self.add_patterns(config.get("exclude"), self.exclude_patterns)
+                continue
+            if (classification := key) in ["unclassified", "pending", "false_positive", "intentional", "bug"]:
+                classification_lower = classification.lower().replace("_", " ")
+                checker = CoverityClassificationChecker(classification=classification_lower, verbose=self.verbose)
+                if isinstance((maximum := config[classification].get("max", 0)), (int, str)):
+                    checker.maximum = int(maximum)
+                if isinstance((minimum := config[classification].get("min", 0)), (int, str)):
+                    checker.minimum = int(minimum)
```

Comment:
Why the type check?

Reply:
It can be removed because I convert it to `int`

Reply:
Resolved in e9e8dbe81d71df5497321c023ed8f7e221476139